backend/persistent/plotly-api.js

```diff
@@ -164,36 +165,92 @@ export function patchGrid(fid, requestor, body) {
     });
 }
 
-export function updateGrid(rows, fid, uids, requestor) {
-    const {username, apiKey, accessToken} = getCredentials(requestor);
-
-    // TODO - Test case where no rows are returned.
-    if (uids.length !== (rows[0] || []).length) {
-        Logger.log(`
-            A different number of columns was returned in the
-            query than what was initially saved in the grid.
-            ${rows[0].length} columns were queried,
-            ${uids.length} columns were originally saved.
-            The connector does not create columns (yet),
-            and so we will only update the first ${uids.length}
-            columns.
-        `);
-    }
+export function updateGrid(rows, columnnames, fid, requestor) {
+    const numColumns = columnnames.length;
+    const columns = getColumns(rows, numColumns);
+    const columnEntries = columns.map((column, columnIndex) => {
+        return { data: column, name: columnnames[columnIndex], order: columnIndex };
+    });
 
-    /*
-     * For now, only update up to the number of columns that are
-     * already saved. In the future, we should just create more
-     * columns. See error message above.
-     */
-    const columns = getColumns(rows, uids.length);
-
-    const url = `grids/${fid}/col?uid=${uids.join(',')}`;
-    const body = {
-        cols: JSON.stringify(columns.map(column => ({
-            data: column
-        })))
-    };
-    return plotlyAPIRequest(url, {body, username, apiKey, accessToken, method: 'PUT'});
+    const {username, apiKey, accessToken} = getCredentials(requestor);
+    const baseUrl = `grids/${fid}/col`;
+    const baseParams = { username, apiKey, accessToken };
+
+    // fetch latest grid to get the source of truth
+    return getGrid(fid, requestor)
```

Comment:
Potentially, this is a very expensive request, since it returns all the grid data, when we are only interested in column names, `uid` and `order`.

@tarzzz Is there any other way to get this information?

Reply:
good point! I believe the API supports GET for the `/col` endpoint

Reply:
The API does support `GET /col`. I can make that update.

Reply:
btw, our API is documented here: https://api.plot.ly/v2/grids/ ..